### ISSUE TITLE
Glowroot with Wildfly Swarm - ClassNotFoundException

### ISSUE THREAD

**Issue by AndreasMundorf** [glowroot/glowroot#299]
Hi,
when I simple work after the docs and add -javaagent:path/to/glowroot.jar (with correct path) to my wildfly swarm application, I get the following exceptions during startup:

```
`java.lang.BootstrapMethodError: call site initialization exception
	at java.lang.invoke.CallSite.makeSite(CallSite.java:341)
	at java.lang.invoke.MethodHandleNatives.linkCallSiteImpl(MethodHandleNatives.java:307)
	at java.lang.invoke.MethodHandleNatives.linkCallSite(MethodHandleNatives.java:297)
	at org.wildfly.swarm.container.runtime.ServerBootstrapImpl.bootstrap(ServerBootstrapImpl.java:114)
	at org.wildfly.swarm.Swarm.start(Swarm.java:399)
	at org.wildfly.swarm.Swarm.main(Swarm.java:734)
Caused by: java.lang.NoClassDefFoundError: org/glowroot/agent/plugin/executor/ExecutorAspect$RunnableEtcMixin
	at sun.misc.Unsafe.defineAnonymousClass(Native Method)
	at java.lang.invoke.InnerClassLambdaMetafactory.spinInnerClass(InnerClassLambdaMetafactory.java:326)
	at java.lang.invoke.InnerClassLambdaMetafactory.buildCallSite(InnerClassLambdaMetafactory.java:194)
	at java.lang.invoke.LambdaMetafactory.metafactory(LambdaMetafactory.java:304)
	at java.lang.invoke.CallSite.makeSite(CallSite.java:302)
	... 5 more
Caused by: java.lang.ClassNotFoundException: org.glowroot.agent.plugin.executor.ExecutorAspect$RunnableEtcMixin from [Module "org.wildfly.swarm.container:runtime" from BootModuleLoader@1d207fad for finders [BootstrapClasspathModuleFinder, BootstrapModuleFinder(org.wildfly.swarm.bootstrap:main), ClasspathModuleFinder, ContainerModuleFinder(swarm.container:main), ApplicationModuleFinder(swarm.application:main), org.wildfly.swarm.bootstrap.modules.DynamicModuleFinder@72ed9aad]]
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:198)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:363)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:351)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:93)
	... 10 more
```

`
Has anybody an idea whats missing here? I did not find any hints in stackoverflow or glowroot.org.
Thanks for your help!
Andreas

**Comment by trask** [comment]
Hi @AndreasMundorf, thanks for reporting this!  I'm able to reproduce and will look into it.

**Comment by trask** [comment]
I think this should be resolved now in the latest snapshot release [glowroot-agent-0.9.29-SNAPSHOT-dist.zip](https://oss.sonatype.org/service/local/artifact/maven/redirect?r=snapshots&g=org.glowroot&a=glowroot-agent&c=dist&e=zip&v=0.9.29-SNAPSHOT).

**Comment by Sohailjp** [comment]
Hi,

I am facing the same issue even after using the agent  glowroot-agent-0.9.29-SNAPSHOT-dist.zip. I am using JBoss EAP 7.0.8, running liferay portal DXP


18:56:15,788 ERROR [Framework Event Dispatcher: Equinox Container: 003fb839-caf7-0017-1993-e970c6d29282][org_eclipse_equinox_console:97] Fram
eworkEvent ERROR 
org.osgi.framework.BundleException: Exception in org.eclipse.equinox.console.command.adapter.Activator.start() of bundle org.eclipse.equinox.
console.
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:795)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:724)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:951)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:328)
        at org.eclipse.osgi.container.Module.doStart(Module.java:566)
        at org.eclipse.osgi.container.Module.start(Module.java:434)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1582)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1562)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1533)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1476)
        at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
        at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)
Caused by: org.osgi.framework.BundleException: Error starting module.
        at org.eclipse.osgi.container.Module.doStart(Module.java:575)
        at org.eclipse.osgi.container.Module.start(Module.java:434)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.start(EquinoxBundle.java:402)
        at org.eclipse.equinox.console.command.adapter.Activator.startBundle(Activator.java:247)
        at org.eclipse.equinox.console.command.adapter.Activator.start(Activator.java:237)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:774)
        at org.eclipse.osgi.internal.framework.BundleContextImpl$3.run(BundleContextImpl.java:1)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:767)
        ... 12 more
Caused by: java.lang.NoClassDefFoundError: org/glowroot/agent/plugin/executor/ExecutorAspect$RunnableEtcMixin
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.defineClass(ModuleClassLoader.java:272)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.defineClass(ClasspathManager.java:632)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findClassImpl(ClasspathManager.java:588)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClassImpl(ClasspathManager.java:540)
        at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:527)
        at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:324)
        at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:330)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:407)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:357)
        at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:349)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.loadBundleActivator(BundleContextImpl.java:757)
        at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:711)
        at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:951)
        at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:328)
        at org.eclipse.osgi.container.Module.doStart(Module.java:566)
        ... 20 more

**Comment by trask** [comment]
Hi @Sohailjp, thanks for reporting this.  I think it should be fixed now by 3be103e4c0fd0639c8da55a06bffbf720e3d3c20 in the latest snapshot release [glowroot-agent-0.10.0-SNAPSHOT-dist.zip](https://oss.sonatype.org/service/local/artifact/maven/redirect?r=snapshots&g=org.glowroot&a=glowroot-agent&c=dist&e=zip&v=0.10.0-SNAPSHOT).

**Comment by Sohailjp** [comment]
Thank you very much, its working now.
